lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -141,6 +142,19 @@ def __init__(self, message: str):
         super().__init__(self.message)
 
 
+class NadConfigChangedEvent(EventBase):
+    """Event triggered when an existing network attachment definition is changed."""
+
+    def __init__(self, handle: Handle):
+        super().__init__(handle)
+
+
+class NadConfigChangedCharmEvents(CharmEvents):
+    """NAD config changed events."""
+
+    nad_config_changed = EventSource(NadConfigChangedEvent)
```

Comment:
Custom events are for used by the charm, not the lib. Here it seems like we don't need this custom event at all if the whole behavior is handled in the lib.

Reply:
Custom event is removed.